File: src/main/java/com/sequenceiq/cloudbreak/domain/CbUserRole.java
package com.sequenceiq.cloudbreak.domain;

public enum CbUserRole {
    DEPLOYER("deployer"),
    ADMIN("admin"),
    USER("user");

    private final String value;

    private CbUserRole(String value) {
        this.value = value;
    }

    public static CbUserRole fromString(String text) {
        if (text != null) {
            for (CbUserRole cbUserRole : CbUserRole.values()) {
                if (text.equalsIgnoreCase(cbUserRole.value)) {
                    return cbUserRole;
                }
            }
        }
        return null;
    }

}

File: cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/type/CbUserRole.kt
package com.sequenceiq.cloudbreak.common.type

enum class CbUserRole private constructor(private val value: String) {
    DEPLOYER("deployer"),
    ADMIN("admin"),
    USER("user");


    companion object {

        fun fromString(text: String?): CbUserRole? {
            if (text != null) {
                for (cbUserRole in CbUserRole.values()) {
                    if (text.equals(cbUserRole.value, ignoreCase = true)) {
                        return cbUserRole
                    }
                }
            }
            return null
        }
    }

}
